pkg/networkpath/traceroute/tcp/utils.go

```diff
@@ -8,21 +8,14 @@ package tcp
 import (
 	"fmt"
 	"net"
-	"syscall"
 
 	"github.com/google/gopacket"
 	"github.com/google/gopacket/layers"
 	"golang.org/x/net/ipv4"
 )
 
-type (
-	// tcpResponse encapsulates the data from a
-	// TCP response needed for matching
-	tcpResponse struct {
-		SrcIP       net.IP
-		DstIP       net.IP
-		TCPResponse layers.TCP
-	}
+var (
+	buf = gopacket.NewSerializeBuffer()
```

Comment:
Are we sure this can be a global? Will this not be used simultaneously by different callers? Another option would be to use a `sync.Pool` for these to alleviate any concerns about simultaneous usage.

Reply:
Moved this to the `TCPv4` and `UDPv4` structs respectively